src/components/TimespanForm.js

```diff
@@ -65,7 +67,28 @@ class TimespanForm extends Component {
             beginTime: structuralMetadataUtils.toHHmmss(startTime),
             endTime: structuralMetadataUtils.toHHmmss(endTime)
           },
-          this.updateChildOfOptions()
+          () => {
+            this.updateChildOfOptions();
+            // Prevent from overlapping when dragging the handles
+            if (!this.formIsValid()) {
+              const {
+                startTime,
+                endTime
+              } = waveformDataUtils.preventSegmentOverlapping(
+                segment,
+                peaksInstance.peaks
+              );
+              this.setState(
+                {
+                  beginTime: structuralMetadataUtils.toHHmmss(startTime),
+                  endTime: structuralMetadataUtils.toHHmmss(endTime)
+                },
+                () => {
+                  this.updateChildOfOptions();
+                }
+              );
+            }
+          }
```

Comment:
Does this timing of the `setState` callbacks need to be this way?  Possible to use the `if (!this.formIsValid()) {` as the outer conditional maybe?   Just a thought...

Reply:
Yeah, good point :+1: I will check it and see whether the functionality holds if use the condition outside.

Reply:
@adamjarling It works even without the validity check. I removed the `if` condition totally, and pass the updated segment to `preventSegmentOverlapping()` function. And it returns valid values for the particular segment, when editing segment overlap other segments.

Reply:
@Dananji Great.